samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -1140,48 +1035,67 @@ static bool light_ctl_setunack(struct bt_mesh_model *model,
 			       struct bt_mesh_msg_ctx *ctx,
 			       struct net_buf_simple *buf)
 {
-	u8_t tid;
-	u16_t lightness, temp, delta_uv;
+	u8_t tid, buf_len, delay;
+	s16_t delta_uv;
+	u16_t lightness, temp;
+	s64_t now;
 	struct net_buf_simple *msg = model->pub->msg;
 	struct light_ctl_state *state = model->user_data;
 
+	buf_len = buf->len;
 	lightness = net_buf_simple_pull_le16(buf);
 	temp = net_buf_simple_pull_le16(buf);
-	delta_uv = (int16_t) net_buf_simple_pull_le16(buf);
+	delta_uv = (s16_t) net_buf_simple_pull_le16(buf);
 	tid = net_buf_simple_pull_u8(buf);
 
-	if (state->last_tid == tid && state->last_tx_addr == ctx->addr) {
-		if (temp < TEMP_MIN || temp > TEMP_MAX) {
-			return false;
-		}
+	if (temp < TEMP_MIN || temp > TEMP_MAX) {
+		return false;
+	}
+
+	now = k_uptime_get();
+	if (state->last_tid == tid && state->last_tx_addr == ctx->addr &&
+	    (now - state->last_msg_timestamp <= 6000)) {
 		return true;
 	}
 
-	if (temp < TEMP_MIN || temp > TEMP_MAX) {
+	switch (buf_len) {
+	case 0x07: /* Target States (2+2+2) + TID (1) */
+		state->tt = 0;
+		delay = 0;
+		state->tt_counter = 0;
+		break;
+	case 0x09: /* Target States (2+2+2) + TID (1) + TT (1) + Delay (1) */
+		state->tt = net_buf_simple_pull_u8(buf);
+		delay = net_buf_simple_pull_u8(buf);
+		break;
+	default:
 		return false;
 	}
 
 	state->last_tid = tid;
 	state->last_tx_addr = ctx->addr;
+	state->last_msg_timestamp = now;
+	state->target_lightness = lightness;
 
 	if (temp < state->temp_range_min) {
 		temp = state->temp_range_min;
 	} else if (temp > state->temp_range_max) {
 		temp = state->temp_range_max;
 	}
 
-	state->lightness = lightness;
-	state->temp = temp;
-	state->delta_uv = delta_uv;
+	state->target_temp = temp;
+	state->target_delta_uv = delta_uv;
 
-	state_binding(CTL, CTL_TEMP);
-	update_light_state();
+	light_ctl_tt_values(state);
+	lightness_transition = LIGTHT_CTL_TT;
+
+	k_timer_start(&lightness_transition_timer, K_MSEC(5*delay),
+		      K_MSEC(100));
```

Comment:
Where does the K_MSEC(100) come from? The model spec? If so, please put a code comment explaining this.

Reply:
I will repair other things which are mostly coding style issues.

But this is important one as of now. As per Mesh Model Specification 3.1.3.1, result
of calculation of transition time is always multiple of 100 milliseconds that is why I used K_MSEC(100). 

if TT = 0x85, then transition time = 10 Secs * 5 = 50 Secs = 50000 millisecs

So counter = 50000 / 100 = 50.

...& delta value which get added after every 100 milliseconds would be = delta/50

Reply:
Now 100 get replaced by Server's **cal_tt**.